container/pull.bzl

```
@@ -210,13 +219,44 @@ def _impl(repository_ctx):
     # foo.digest for an image named foo.
     repository_ctx.symlink(repository_ctx.path("image/digest"), repository_ctx.path("image/image.digest"))
 
+    tar_gz_files = []
+    sha256_files = []
+    config = json.decode(repository_ctx.read(repository_ctx.path("image/config.json")))
+    for i in range(len(config["rootfs"]["diff_ids"])):
+        layer_id = "{}".format(i)
+        layer_id = "000"[0:3-len(layer_id)] + layer_id
+        tar_gz_files.append("{}.tar.gz".format(layer_id))
+        sha256_files.append("{}.sha256".format(layer_id))
+
+    download_target = ""
+    if repository_ctx.attr.lazy_download:
+        later_fetch_args = [
+            "$(location {})".format(puller),
+            "-directory",
+            "$(RULEDIR)",
+        ] + args
+
+        download_target = """
+genrule(
+    name = "download",
+    message = "Fetching",
+    outs = ["{outs}"],
+    cmd = "{args}",
+    tools = ["{tool}"],
```

```suggestion
    tools = ["{tool}"],
    tags = ["requires-network"],
```
to make sure network is allowed even in build sandbox